aip/0180.md

```diff
@@ -0,0 +1,126 @@
+    populated, even if it introduces redundancy.
+- Enums **may** be freely added to request messages. For response messages and
+  resources, be aware that it is possible that user code does not handle new
+  values gracefully. In this situation, enums which are expected to receive new
+  values **should** document this, and while enum values still **may** be added
+  in this situation, appropriate caution **should** be used.
+
+**Note:** It is possible when adding a component closely related to an existing
+component (for example, `string foo_value` when `string foo` already exists) to
+enter a situation where generated code will conflict. Service owners **should**
+be aware of subtleties in the tooling they or their users are likely to use
+(and tool authors **should** endeavor to avoid such subtleties if possible).
+
+### Removing or renaming components
+
+Existing components (interfaces, methods, messages, fields, enums, or enum
+values) **must not** be removed from existing APIs in the same major version.
+Removing a component is a backwards incompatible change.
+
+**Important:** Renaming a component is semantically equivalent to "remove and
+add". In cases where these sorts of changes are desirable, a service **may**
+add the new component, but **must not** remove the existing one.
+
+### Changing the type of fields
+
+Existing fields and messages **must not** have their type changed, even if the
+new type is wire-compatible, because type changes alter generated code in a
+breaking way.
+
+**Note:** An exception to this rule is that `int32` values **may** be changed
+to `int64`.
+
+### Changing resource names
+
+A resource **must not** change its name.
+
+Unlike most breaking changes, this affects major versions as well: in order for
+a client to expect to use v2.0 access a resource that was created in v1.0 or
+vice versa, the same resource name **must** be used in both versions.
+
+More subtly, the set of valid resource names **should not** change either, for
+the following reasons:
+
+- If resource name formats become more restrictive, a request that would
+  previously have succeeded will now fail.
+- If resource name formats become less restrictive than previously documented,
+  then code making assumptions based on the previous documentation could break.
+  Users are very likely to store resource names elsewhere, in ways that may be
+  sensitive to the set of permitted characters and the length of the name.
+  Alternatively, users might perform their own resource name validation to
+  follow the documentation.
+  - For example, Amazon gave customers [a lot of warning][ec2] and had a
+    migration period when they started allowing longer EC2 resource IDs.
+
+### Semantic changes
+
+Code will often depend on API behavior and semantics, _even when such behavior
+is not explicitly supported or documented_. Therefore, APIs **must not** change
+visible behavior or semantics, as such changes will be seen as breaking by
+users.
```

Comment:
Again, this sounds like there's no wiggle room - we should be clearer that there *has* to be wiggle room.

Reply:
Done.